add buildResultProperties to the build report data

dennis-behm opened the pull request:
In case of a build connecting via the repoClient to the dbb web application, we add the build result properties to a generic DBB Record named `DBB.BuildResultProperties`

See sample :
![image](https://user-images.githubusercontent.com/28918869/110509036-f88ef180-8101-11eb-8f32-f79d32f75895.png)

drbruce-git commented on build.groovy:
```diff
@@ -516,11 +502,43 @@ def finalizeBuildProcess(Map args) {
+	def transformer = HtmlTransformer.getInstance()
+	transformer.transform(jsonOutputFile, htmlTemplate, css, renderScript, htmlOutputFile, buildReportEncoding)
+	
+	// save json file
```
Another minor issue.  Can you create and save the data file before the html file?  I know it does not matter functionally but since its passed into the html traqnsformer, logically it should be created and saved before the transformer is called.

drbruce-git commented on build.groovy:
```diff
@@ -473,29 +473,15 @@ def finalizeBuildProcess(Map args) {
 
```
Minor issue. Can you move the build report data file comment and variables down with the html file variables and where you actually create and save the file?

dennis-behm commented:
@drbruce-git
Can you please have a look to the updated PR.
Additionally to your requested changes, I 
- added the giturl into the properties
- moved the println of the update of the buildresult to the end and included the buildreporturl

Many thanks.
